### ISSUE TITLE
Crash on resuming during program run (16.12 regression)

### ISSUE THREAD

**Issue by robhagemans** [robhagemans/pcbasic#30]
`10 for i = 1to 10000:print i: next`

Ctrl+C during run

```
./pcbasic.py --resume
Exception in thread Thread-1:
Traceback (most recent call last):
  File "/usr/lib/python2.7/threading.py", line 801, in __bootstrap_inner
    self.run()
  File "/usr/lib/python2.7/threading.py", line 754, in run
    self.__target(*self.__args, **self.__kwargs)
  File "/home/rob/Projects/basic-project/pc-basic/pcbasic/main.py", line 157, in run_session
    session = state.zunpickle(state_file).attach(iface)
  File "/home/rob/Projects/basic-project/pc-basic/pcbasic/state.py", line 82, in zunpickle
    return pickle.loads(zlib.decompress(f.read()))
TypeError: Required argument 'object' (pos 1) not found
```

**Comment by robhagemans** [comment]
This may have to do with the inability to pickle `memoryview` objects, see https://bugs.python.org/issue27098

**Comment by robhagemans** [comment]
- Does not happen if we simply define a variable `i`. 
- Does not happen if we allow the  program to finish and then exit.
- Does happen if we include a `STOP` statement and exit after the break. 
- Also happens if we interrupt PC-BASIC with `Alt+F4` while the program is running.

**Comment by robhagemans** [comment]
Another suspect is the `ByteStream` class which is derived from `_pyio.BytesIO`

**Comment by robhagemans** [comment]
`10 i=1: goto 10` does not have the issue. `FOR` loops only?

**Comment by robhagemans** [comment]
`interpreter.py` lines 385, 386:
```
        counter_view = self._scalars.view(varname)
        self.for_stack.append((varname, counter_view, stop, step, step.sign(), forpos, nextpos,))
```

This stores a view in a structure that is to be pickled - may be the cause?

**Comment by robhagemans** [comment]
The `memoryview` is the issue. Apparently I'm storing most variables simply as `bytearray`s, but in a `FOR`-loop I'm keeping a copy of the `Value` object, which contains a `memoryview`. Adding `__setstate__` and `__getstate__` to `Value` to convert the `memoryview` to `bytearray` and back avoids the crash, but cuts the connection between the loop variable and the view of it which is being incremented.

**Comment by robhagemans** [comment]
As far as I can see, the loop record contains the `Value` objects `counter_view`, `stop` and `step`. Not sure why the `Scalars` object itself pickles without problems in the absence of a `FOR`-loop.

**Comment by robhagemans** [comment]
Fixed by 35e5640
